testsuite/testcases/src/lib.rs

```diff
@@ -168,28 +201,44 @@ impl NetworkTest for dyn NetworkLoadTest {
             .build()
             .map_err(|err| anyhow!("Failed to start runtime for transaction emitter. {}", err))?;
 
-        let job = rt
-            .block_on(emitter.start_job(ctx.swarm().chain_info().root_account, emit_job_request))?;
-        info!("Starting emitting txns for {} secs", duration.as_secs());
+        let job =
+            rt.block_on(emitter.start_job(swarm.chain_info().root_account, emit_job_request, 3))?;
 
-        self.test(ctx.swarm(), duration)?;
+        let warmup_duration = duration.mul_f32(warmup_duration_fraction);
+        let cooldown_duration = duration.mul_f32(cooldown_duration_fraction);
+        let test_duration = duration - warmup_duration - cooldown_duration;
+        info!("Starting emitting txns for {}s", duration.as_secs());
 
-        info!("Ran for {} secs, stopping job...", duration.as_secs());
-        let txn_stat = rt.block_on(emitter.stop_job(job));
-        info!("Stopped job");
+        std::thread::sleep(warmup_duration);
+        info!("{}s warmup finished", warmup_duration.as_secs());
 
-        ctx.report
-            .report_txn_stats(self.name().to_string(), &txn_stat, duration);
+        job.start_next_phase();
 
-        let end_timestamp = SystemTime::now()
-            .duration_since(UNIX_EPOCH)
-            .expect("Time went backwards")
-            .as_secs();
+        let test_start = Instant::now();
+        self.test(swarm, test_duration)?;
+        let actual_test_duration = test_start.elapsed();
+        info!(
+            "{}s test finished after {}s",
+            test_duration.as_secs(),
+            actual_test_duration.as_secs()
+        );
 
-        ctx.check_for_success(&txn_stat, &duration)?;
+        job.start_next_phase();
 
-        self.finish(ctx.swarm(), start_timestamp, end_timestamp)?;
+        std::thread::sleep(cooldown_duration);
```

Comment:
What is special we are doing in cool down that we need it as a separate phase? It seems like we should first stop the job and gather the stats for stopping period?

Reply:
when we stop the job, we stop sending new traffic. without new traffic it is easier for nodes to finish what they have in mempool, and latencies / drop rate might be smaller. I don't have a long cooldown period, 5%, compared to 20% for the warmup (for my tests. default it both being 5% for landblocking)

Reply:
That makes sense, but my point is why do we need to sleep here? You have not stopped the traffic yet, sleeping here is not going to give you the cool down stats you are looking for? May be you can just remove the sleep, stop the job and measure the cool down stats?

Reply:
Stats are measured on a per- transaction basis.  So for transactions submitted before start_new_phase(), their completion/ latency is going to be written later but into an old phase. That seems like a way to make things representative.

So stats for main phase are still being collected during the cooldown phase, and why we delay the stopping of traffic, because it will affect latency of already submitted transactions

Reply:
Ah, I see. Makes sense